src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -125,11 +131,15 @@ public SwerveDriveKinematics getKinematics() {
     public void periodic() {
         // Update pose estimator from swerve module states
         Rotation2d gyroAngle = getGyroHeading();
-        poseEstimator.update(
+        Pose2d estimate = poseEstimator.update(
             gyroAngle,
             getModuleStates()
         );
 
+        // Add vision pose estimate to pose estimator
+        var res = vision.getRobotPose(estimate);
```

Comment:
why 'var'? why not just write the type :( i can't see inferences on web github

is this a blocking call? does this wait for communications over networking? if so, we shouldn't tie up periodic waiting for this to happen

Reply:
I used `var` because I didn't want to write `Pair<Pose2d, Double>` :D the example code I copied from PhotonVision also uses `var` for some nasty types (like I think the pose estimator itself returns like `Optional<Pair<Pose3d, Double>>` or smth); it's also as far as I can tell not a blocking call in the same way the odometry pose estimator update is not a blocking call.